utils/kubernetes/service.go

```diff
@@ -73,5 +75,20 @@ func GetEndpoint(ctx context.Context, opts *ServiceOptions, obj *corev1.Service)
 		}
 	}
 
+	if !utils.TcpCheck(endpoint.External) {
+		if len(strings.SplitAfter(opts.APIServerURL, "://")) > 1 {
+			endpoint.External.Address = strings.SplitAfter(strings.SplitAfter(opts.APIServerURL, "://")[1], ":")[0]
+			if len(endpoint.External.Address) > 0 {
+				endpoint.External.Address = endpoint.External.Address[:len(endpoint.External.Address)-1]
+			}
+		}
+		if !utils.TcpCheck(endpoint.External) {
```

Comment:
check whether looking at `APIServerURL` return something alive...

Reply:
It wouldn't matter correct? As long as the port responds, correct me if I'm wrong. @mgfeller

Reply:
no, it wouldn't, I was just "thinking aloud" :smile: